### PR TITLE
chore: stricter sidebar interface

### PR DESCRIPTION
**Pull request by wegry**
As part of prep work for reworking mole & sidebar interactions, go through the existing sidebar interface and look for non-public usages that could be private.

- chore: true private in several @internal classes
- remove several `any`s

### CODE REVIEW

**Comment by wegry** [comment]
As far as class methods and fields moved from non-public to private go, to check the move, I first pulled properties moved over with:

```sh
git show -c f8c8a3d7 | rg --only-matching '\b_(\w+)' -r '$1' | sort | uniq | rg --only-matching -U '(\w+)(\n)?' -r '$1|' | tr -d '\n'
```

Based on this list I ran, to check for usages that might've been obscured by `any`

```
rg '\b_(addButton|addonSidebarHiddenChanged|appId|appSidebarRefSetter|appSidebarView|butterBar|butterBarDriver|closeButton|closeSidebarAndDeactivateButton|companionSidebarContentContainerEl|companionSidebarOuterWrapper|composeViewDriverStream|containerProp|contentContainers|createThreadSidebarIfNeeded|currentIds|currentRouteViewDriver|customListRouteIDs|customListSearchStringsToRouteIds|customRouteIDs|driver|envData|getActiveButtonContainer|getShouldGlobalAppSidebarOpen|getShouldThreadAppSidebarOpen|globalButtonContainers|globalIconArea|gmailRouteProcessor|instanceId|instanceIdsToDescriptors|keyboardShortcutHelpModifier|lastCustomThreadListActivity|logger|messageIDsToThreadIDs|messageViewDriverStream|navMarkerHiddenChanged|openSidebarAndActivateButton|opts|orderManager|page|pageCommunicator|pageCommunicatorPromise|recentSyncDraftIds|removeButton|renderThreadSidebar|renderThreadSidebarIfPresent|routeViewDriverStream|rowListViewDriverStream|selectedThreadRows|setShouldGlobalAppSidebarOpen|setShouldThreadAppSidebarOpen|setupComposeViewDriverStream|setupElement|setupEventStreams|setupMessageViewDriverStream|setupRouteSubViewDriver|setupThreadRowViewDriverKefirStream|setupToolbarViewDriverStream|stopper|threadButtonContainers|threadIconArea|threadRowIdentifier|threadRowViewDriverKefirStream|threadRowViewSelectionChanges|threadSidebarComponent|threadSidebarContainerEl|threadViewDriverLiveSet|timestampAccountSwitcherReady|timestampGlobalsFound|timestampOnready|toolbarViewDriverLiveSet|updateHighlightedAppThreadIconBus|userInfo|xhrInterceptorStream)\b'
```

There are several properties which have non-public usages elsewhere:

```
_addButton
_appId
_closeButton
_closeSidebarAndDeactivateButton
_driver
_gmailRouteProcessor
_instanceId
_opts
_page
_recentSyncDraftIds
_renderThreadSidebar
_setupElement
_stopper
_threadSidebarContainerEl
```

File: src/platform-implementation-js/dom-driver/gmail/views/gmail-app-sidebar-view/index.ts [hunk 1]
```diff
@@ -9,106 +9,70 @@ import type GmailDriver from '../../gmail-driver';
...
-
-  _setShouldGlobalAppSidebarOpen(open: boolean) {
-    try {
-      global.localStorage.setItem(
-        'inboxsdk__global_app_sidebar_should_open',
-        String(open),
-      );
-    } catch (err) {
-      console.error('error saving', err);
-    }
```
Comment (wegry) [inline src/platform-implementation-js/dom-driver/gmail/views/gmail-app-sidebar-view/index.ts:99]: This class isn't part of public interface.

Based on grepping, it appears that this code was moved into src/platform-implementation-js/dom-driver/gmail/views/gmail-app-sidebar-view/primary/index.tsx.

File: src/platform-implementation-js/driver-common/sidebar/ContentPanelViewDriver.ts [hunk 2]
```diff
@@ -20,47 +20,48 @@ export interface ContentPanelDescriptor {
...
-    this._isGlobal = Boolean(isGlobal);
-    this._stopper = this._eventStream
+    this.#driver = driver;
+    this.#sidebarId = sidebarId;
+    this.#isGlobal = Boolean(isGlobal);
+    this.#stopper = this.#eventStream
       .ignoreValues()
       .beforeEnd(() => null)
       .toProperty();
-    const document = global.document; //fix for unit test
```
Comment (wegry) [inline src/platform-implementation-js/driver-common/sidebar/ContentPanelViewDriver.ts:46]: This line was removed.

Comment (dmy7r0) [reply src/platform-implementation-js/driver-common/sidebar/ContentPanelViewDriver.ts:46]: It's a bit unclear what this means. Does this removal mean unit tests just work without it or we aren't referencing the file in tests?

Comment (wegry) [reply src/platform-implementation-js/driver-common/sidebar/ContentPanelViewDriver.ts:46]: Unit tests work without this line. I suspect it was probably to account for an old version of `jsdom` or maybe even the pre-jest SDK test library (mocha?).

**Pull request merged**